### DISCUSSION TITLE
Carry-Over Threshold

### DISCUSSION THREAD

**Discussion by 73** [jotaen/klog#220] [Feature ideas & discussions]
Hi, 

it would be nice if I were able to set a monthly threshold for carry-over positive diff in time from month to month. Some working contracts say you can only carry hours over to the next month that are more than X. 

I don't know if I made myself clear so here an example:
```
            Total    Should     Diff
2022 Jun  101h45m      91h!  +10h45m -> carry 5h45m
     Jul     142h     147h!      -5h -> carry -5h
     Aug   96h45m      91h!   +5h45m -> carry 45m
     Sep  107h15m     112h!   -4h45m -> carry -4h45m
     Oct     139h     133h!      +6h -> carry 1h
         ======== ========= ========
          586h45m     574h!  +12h45m
```

Currently I do this manually by setting a correction entry on the last of the month for exact time.

Maybe someone else needs that, too?

**Comment by jotaen** [comment]
Thanks for sharing your use-case! I think, your solution to manually do the correction via a dedicated entry/record is the current way to go here. But it seems as if the whole topic around managing properties on a weekly/monthly basis is a recurring theme. Out of curiosity, do you use one file per month, or do you put everything in one big file?

Related discussions that come to mind, just for reference:

- https://github.com/jotaen/klog/discussions/198
- https://github.com/jotaen/klog/discussions/212

**Comment by 73** [comment]
Hi @jotaen 

Thanks for getting back to me.
I place everything in one big file. `klog` is after half a year still very fast for me. I thought of using one file per year. I guess I have to carry some diff over when changing files. We will see how that goes when I get to that.

**Reply by jotaen** [reply]
Since you mentioned performance, it reminded me that I didn’t check klog’s performance in quite a while. I just added [an automated benchmark to the repo here](https://github.com/jotaen/klog/pull/223), so that it’s easier to keep an eye on that aspect.
